### ISSUE TITLE
Feature Change: New notebook prompts for either .dib or .ipynb

### ISSUE THREAD

**Issue by brettfo** [dotnet/interactive#1305]
tl;dr
=====

The behavior of the ".NET Interactive: Create new blank notebook" command is changing.  The old behavior was to create, e.g., `Untitled-1.ipynb`; the new behavior will then ask the user which notebook type they'd prefer, so they'll either get, e.g., `Untitled-1.dib` or `Untitled-1.ipynb`.

Why?
====

The VS Code notebook APIs are still in preview and are considered unstable, and because of this, it's become really difficult to coordinate breaking changes and user experiences between VS Code, the Jupyter extension (the handler for .ipynb files), and us.  To reduce the complexity, we're adding the possibility to use .dib as a new untitled notebook and that's simply because we fully own that extension so we know we'll always work, but we don't want to too badly mess up existing users' workflows who prefer .ipynb, hence the choice.

This...

![image](https://user-images.githubusercontent.com/926281/116148472-cf1c2c00-a695-11eb-8449-388320dc077a.png)

...now leads to this...

![image](https://user-images.githubusercontent.com/926281/116148505-d7746700-a695-11eb-9c14-1ffb3ebcd28d.png)

**Comment by brettfo** [comment]
PR #1304 implements this change.
